app/javascript/packs/application.js

```diff
@@ -17,3 +17,11 @@ require('channels')
 
 require('trix')
 require('@rails/actiontext')
+
+document.addEventListener('DOMContentLoaded', function (event) {
+  document
+    .querySelector('.alert button.close')
```

Comment:
Pretty sure you have a reason for this but... Why you can not attach the click on the component itself and instead you have to wait for "DOMContentLoaded"?

Reply:
I'm not sure if this is an actual issue, but for correctness is should be done this way. If the element isn't present on the first load (this comes from the server so it should be there before window.onload triggers) it will fail to bind the click event.